pkg/internal/ebpf/common/kafka_detect_transform.go

```diff
@@ -256,6 +296,24 @@ func getTopicOffsetFromFetchOperation(header *Header) int {
 	return offset
 }
 
+func readUnsignedVarint(data []byte) (int, error) {
```

Comment:
Do you mind adding unit tests for this function? Encode some larger numbers, make sure they are decoded correctly.